File: app/src/main/java/tiiehenry/android/app/snapshotor/config/SyncConfig.java
package tiiehenry.android.app.snapshotor.config;

import com.alibaba.fastjson2.JSON;

import java.util.HashSet;
import java.util.Set;

/**
 * 同步配置类
 * 纯数据类，使用 fromJson/toJson 进行序列化/反序列化
 */
public class SyncConfig {
    // 配置字段，直接访问
    public boolean enableSyncToTarget = false;
    public Set<String> syncTargets = new HashSet<>();
    public boolean enableSyncToSystem = false;
    public Set<String> syncSystems = new HashSet<>();
    public int syncType = 0;

    /**
     * 从 JSON 字符串解析配置（静态工厂方法）
     */
    public static SyncConfig fromJson(String jsonString) {
        return JSON.parseObject(jsonString, SyncConfig.class);
    }

    /**
     * 将配置转换为 JSON 字符串
     */
    public String toJson() {
        return JSON.toJSONString(this);
    }
}

File: app/src/main/java/tiiehenry/android/app/snapshotor/config/SyncConfig.kt
package tiiehenry.android.app.snapshotor.config

import com.tencent.mmkv.MMKV

class SyncConfig(private val mmkv: MMKV) {
    private companion object {
        const val KEY_ENABLE_SYNC_TO_TARGET = "enableSyncToTarget"
        const val KEY_SYNC_TARGETS = "syncTargets"
        const val KEY_ENABLE_SYNC_TO_SYSTEM = "enableSyncToSystem"
        const val KEY_SYNC_SYSTEMS = "syncSystems"
        const val KEY_SYNC_TYPE = "syncType"
        const val KEY_UNINSTALL_ARCHIVED = "uninstallArchived"
    }

    var enableSyncToTarget: Boolean
        get() = mmkv.decodeBool(KEY_ENABLE_SYNC_TO_TARGET)
        set(value) {
            mmkv.encode(KEY_ENABLE_SYNC_TO_TARGET, value)
        }

    var syncTargets: Set<String>
        get() = mmkv.decodeStringSet(KEY_SYNC_TARGETS, emptySet()) ?: emptySet()
        set(value) {
            mmkv.encode(KEY_SYNC_TARGETS, value)
        }

    var enableSyncToSystem: Boolean
        get() = mmkv.decodeBool(KEY_ENABLE_SYNC_TO_SYSTEM)
        set(value) {
            mmkv.encode(KEY_ENABLE_SYNC_TO_SYSTEM, value)
        }

    var syncSystems: Set<String>
        get() = mmkv.decodeStringSet(KEY_SYNC_SYSTEMS)?:emptySet()
        set(value) {
            mmkv.encode(KEY_SYNC_SYSTEMS, value)
        }

    var syncType: Int
        get() = if (mmkv.containsKey(KEY_SYNC_TYPE)) mmkv.decodeInt(KEY_SYNC_TYPE) else 0
        set(value) {
            mmkv.encode(KEY_SYNC_TYPE, value)
        }
}